src/main/scala/mesosphere/marathon/event/http/HttpEventStreamActor.scala

```diff
@@ -0,0 +1,65 @@
+package mesosphere.marathon.event.http
+
+import akka.actor.Actor
+import akka.event.EventStream
+import mesosphere.marathon.api.v2.json.Formats._
+import mesosphere.marathon.event._
+import mesosphere.marathon.event.http.HttpEventStreamActor._
+import org.apache.log4j.Logger
+import play.api.libs.json.Json
+
+/**
+  * A HttpEventStreamHandle is a reference to the underlying client http stream.
+  */
+trait HttpEventStreamHandle {
+  def connectionOpen: Boolean
+  def remoteAddress: String
+  def sendMessage(message: String): Unit
+}
+
+/**
+  * This actor handles subscriptions from event stream handler.
+  * It subscribes to the event stream and pushes all marathon events to all listener.
+  * @param eventStream the marathon event stream
+  */
+class HttpEventStreamActor(eventStream: EventStream) extends Actor {
+  private[this] var clients = Vector.empty[HttpEventStreamHandle]
```

Comment:
This works by reference identity. 
I thought your idea of using a set refers more to the point of multiple registrations from the same source.